mailet/standard/src/test/java/org/apache/james/transport/matchers/HasExceptionTest.java

```diff
@@ -36,93 +36,126 @@
+                .build();
 
-        matcher.init(matcherConfig);
+        testee.init(matcherConfig);
 
-        assertThat(matcher.match(mockedMail)).isEmpty();
+        assertThat(testee.match(mockedMail)).isEmpty();
     }
 
     @Test
     public void matchShouldReturnEmptyWhenNoExceptionHasOccurred() throws MessagingException {
-        matcher.init(FakeMatcherConfig.builder().matcherName("HasException").condition("java.lang.Exception").build());
+        FakeMatcherConfig matcherConfig = FakeMatcherConfig.builder()
+                .matcherName("HasException")
+                .condition("java.lang.Exception")
+                .build();
 
-        assertThat(matcher.match(mockedMail)).isEmpty();
+        testee.init(matcherConfig);
+
+        assertThat(testee.match(mockedMail)).isEmpty();
     }
 
     @Test
     public void matchShouldReturnEmptyWhenNonExceptionIsAttached() throws MessagingException {
         mockedMail.setAttribute(Mail.MAILET_ERROR_ATTRIBUTE_NAME, new java.lang.String());
 
-        FakeMatcherConfig matcherConfig = FakeMatcherConfig.builder().matcherName("HasException")
-                .condition("java.lang.Exception").build();
+        FakeMatcherConfig matcherConfig = FakeMatcherConfig.builder()
+                .matcherName("HasException")
+                .condition("java.lang.Exception")
+                .build();
 
-        matcher.init(matcherConfig);
+        testee.init(matcherConfig);
 
-        assertThat(matcher.match(mockedMail)).isEmpty();
+        assertThat(testee.match(mockedMail)).isEmpty();
     }
 
     @Test
     public void initShouldRaiseMessagingExceptionWhenInvalidClassName() throws MessagingException {
-        FakeMatcherConfig matcherConfig = FakeMatcherConfig.builder().matcherName("HasException")
-                .condition("java.lang.InvalidClassName").build();
+        FakeMatcherConfig matcherConfig = FakeMatcherConfig.builder()
+                .matcherName("HasException")
+                .condition("java.lang.InvalidClassName")
+                .build();
 
-        assertThatThrownBy(() -> matcher.init(matcherConfig)).isInstanceOf(MessagingException.class);
+        assertThatThrownBy(() -> testee.init(matcherConfig)).isInstanceOf(MessagingException.class);
     }
 
     @Test
     public void initShouldRaiseMessagingExceptionWhenClassNameIsNotException() throws MessagingException {
-        FakeMatcherConfig matcherConfig = FakeMatcherConfig.builder().matcherName("HasException")
-                .condition("java.lang.String").build();
+        FakeMatcherConfig matcherConfig = FakeMatcherConfig.builder()
+                .matcherName("HasException")
+                .condition("java.lang.String")
+                .build();
+
+        assertThatThrownBy(() -> testee.init(matcherConfig)).isInstanceOf(MessagingException.class);
+    }
+    
+    @Test
+    public void initShouldRaiseMessagingExceptionWhenClassNameIsNotFullyQualified() throws MessagingException {
```

Comment:
It seems that this method title is wrong.
What I understand from the test is that a short class name is well recognized, then the _match_ method of the mailer is working.
I propose that you change this method name to _matchShouldReturnAddressesWhenExceptionHasOccurredAndMatchesShortClassName_

BTW thank you for taking care of my comments.

Reply:
Oops, the method name is correct, but the assertion was wrong, and needs to be fixed.